x-pack/plugins/infra/server/usage/usage_collector.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraNodeType } from '../../common/graphql/types';
+import { KbnServer } from '../kibana.index';
+
+const KIBANA_REPORTING_TYPE = 'infraops';
+
+interface InfraopsSum {
+  infraopsHosts: number;
+  infraopsDocker: number;
+  infraopsKubernetes: number;
+  logs: number;
+}
+
+export class UsageCollector {
+  public static getUsageCollector(server: KbnServer) {
+    const { collectorSet } = server.usage;
+
+    return collectorSet.makeUsageCollector({
+      type: KIBANA_REPORTING_TYPE,
+      fetch: async () => {
+        return this.getReport();
+      },
+    });
+  }
+
+  public static countNode(nodeType: InfraNodeType) {
+    const bucket = this.getBucket();
+    this.maybeInitializeBucket(bucket);
+
+    switch (nodeType) {
+      case InfraNodeType.pod:
+        this.counters[bucket].infraopsKubernetes += 1;
+        break;
+      case InfraNodeType.container:
+        this.counters[bucket].infraopsDocker += 1;
+        break;
+      default:
+        this.counters[bucket].infraopsHosts += 1;
+    }
+  }
+
+  public static countLogs() {
+    const bucket = this.getBucket();
+    this.maybeInitializeBucket(bucket);
+    this.counters[bucket].logs += 1;
+  }
+
+  private static counters: any = {};
```

Comment:
Given that we know which counters there are exactly, how about typing them correctly?

Reply:
I'd prefer to do that in a separate cleanup PR.